server/src/service.rs

```diff
@@ -293,4 +296,52 @@ impl Service {
 
         Ok(())
     }
+
+    pub async fn set_locked(
+        &self,
+        locked: bool,
+        objects: &[OwnedObjectPath],
+    ) -> Result<(Vec<OwnedObjectPath>, Vec<OwnedObjectPath>), ServiceError> {
+        let mut locked_changed = Vec::new();
+        let mut locked_unchanged = Vec::new();
+        let collections = self.collections.lock().await;
+
+        for object in objects {
+            for collection in collections.iter() {
+                if object == collection.path() {
+                    if locked == collection.is_locked().await {
```

Comment:
Get the collection locked state once, store it in a variable and use it here and below

Reply:
Done, thanks!